server.js

```diff
@@ -1,34 +1,119 @@
-// TASK 3 : Use mongoose instead of mongodb driver
+//Create Document for slam book
+const http = require('http');
+const url = require('url');
+const SlamBook = require('./model');
 
-const http = require("http");
-const mongoose = require("mongoose");
-
-const PORT = 8000;
+const server = http.createServer((req, res) => {
+  const parsedUrl = url.parse(req.url, true);
+  const endpoint = parsedUrl.pathname;
+  const method = req.method;
+  switch (method) {
+    case 'GET':
+      handleGetRequest(endpoint, parsedUrl, res);
+      break;
+    case 'POST':
+      handlePostRequest(endpoint, req, res);
+      break;
+    case 'PUT':
+      handlePutRequest(endpoint, req, res);
+      break;
+    case 'DELETE':
+      handleDeleteRequest(endpoint, res);
+      break;
+    default:
+      res.writeHead(405, { 'Content-Type': 'text/plain' });
+      res.end('Method Not Allowed');
+  }
+});
 
-const uri =
-  "mongodb+srv://unnatig1920:unnatigupta@cluster0.uokvn7o.mongodb.net/";
+const handlePostRequest = (endpoint, req, res) => {
+  if (endpoint === '/slambook') {
+    let body = '';
+    req.on('data', chunk => {
+      body += chunk.toString();
+    });
 
-async function startServer() {
-  try {
-    await mongoose.connect(uri, {
-      useNewUrlParser: true,
-      useUnifiedTopology: true,
+    req.on('end', () => {
+      const entryData = JSON.parse(body);
+      const newEntry = SlamBook.create(entryData);
+      res.writeHead(201, { 'Content-Type': 'application/json' });
+      res.end(JSON.stringify(newEntry));
     });
-    console.log("Connected to MongoDB");
+  } else {
+    res.writeHead(404, { 'Content-Type': 'text/plain' });
+    res.end('Not Found');
+  }
+};
+const handleGetRequest = (endpoint, parsedUrl, res) => {
+  if (endpoint === '/slambook') {
+    if (parsedUrl.pathname === '/slambook' && !parsedUrl.query.id) {
+      const entries = SlamBook.find();
+      res.writeHead(200, { 'Content-Type': 'application/json' });
+      res.end(JSON.stringify(entries));
+    } else if (parsedUrl.pathname === '/slambook' && parsedUrl.query.id) {
+      const entry = SlamBook.findById(parsedUrl.query.id);
+      if (entry) {
+        res.writeHead(200, { 'Content-Type': 'application/json' });
+        res.end(JSON.stringify(entry));
+      } else {
+        res.writeHead(404, { 'Content-Type': 'text/plain' });
+        res.end('Entry Not Found');
+      }
+    }
+  } else {
+    res.writeHead(404, { 'Content-Type': 'text/plain' });
+    res.end('Not Found');
+  }
+};
+
+const handlePutRequest = (endpoint, req, res) => {
+  if (endpoint.startsWith('/slambook/')) {
+    const entryId = endpoint.split('/').pop();

```

Comment:
The status code in line 84 should be 201 instead of 200, as it is a successful update operation.